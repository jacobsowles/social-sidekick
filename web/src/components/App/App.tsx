import { Auth0Error, Auth0UserProfile } from 'auth0-js';
import React, { Component, Dispatch } from 'react';
import { connect } from 'react-redux';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { Action } from 'redux';

import { setError } from '@actions/error.actions';
import { fetchUserSuccess } from '@actions/user.actions';
import Layout from '@components/Layout';
import AuthService from '@core/auth';
import { init as initIcons } from '@core/icons';
import { AppState } from '@core/types';

initIcons();

interface AppDispatchProps {
  fetchUser: () => void;
}

interface AppStateProps {
  error?: string;
  user?: Auth0UserProfile;
}

type AppProps = RouteComponentProps & AppDispatchProps & AppStateProps;

class App extends Component<AppProps, AppState> {
  public componentDidMount() {
    const authService: AuthService = new AuthService();
    if (authService.isAuthenticated() && !this.props.user) {
      this.props.fetchUser();
    }
  }

  public render() {
    return <Layout user={this.props.user} />;
  }
}

const mapDispatchToProps = (dispatch: Dispatch<Action>): AppDispatchProps => {
  return {
    fetchUser: () => {
      const authService = new AuthService();
      authService.fetchUser((error: Auth0Error, user: Auth0UserProfile) => {
        if (user) {
          dispatch(fetchUserSuccess(user));
        } else {
          dispatch(setError(error.error));
        }
      });
    }
  };
};

const mapStateToProps = (state: AppState): AppStateProps => {
  return {
    error: state.error,
    user: state.user
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(App)
);
