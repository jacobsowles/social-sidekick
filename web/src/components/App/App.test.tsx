import React from 'react';
import { Provider } from 'react-redux';

import { mockStore, init, shallow, toJson } from '@tests/component-test-base';
import App from '@components/App';

describe('App', () => {
  init(
    () => (
      <Provider store={mockStore({})}>
        <App />
      </Provider>
    ),
    'div'
  );

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
