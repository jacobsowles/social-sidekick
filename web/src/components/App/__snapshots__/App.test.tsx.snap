// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component without crashing 1`] = `<WithAlert(withRouter(Connect(App))) />`;
